geode-core/src/distributedTest/java/org/apache/geode/cache/ConnectionPoolDUnitTest.java

```diff
@@ -1298,75 +1311,70 @@ public void run2() throws CacheException {
       CertifiableTestCacheListener<Object, Object> ctl =
           (CertifiableTestCacheListener<Object, Object>) region.getAttributes()
               .getCacheListeners()[0];
+
       for (int i = 0; i < 10; i++) {
         Object key = i;
         ctl.waitForDestroyed(key);
         Region.Entry entry = region.getEntry(key);
         assertThat(entry).isNull();
       }
-      {
-        List<CacheEvent<Object, Object>> list = ctl.getEventHistory();
-        assertThat(10).isEqualTo(list.size());
-        for (int i = 0; i < 10; i++) {
-          Object key = i;
-          EntryEvent ee = (EntryEvent) list.get(i);
-          assertThat(ee.getKey()).isEqualTo(key);
-          assertThat(ee.getOldValue()).isNull();
-          assertThat(Operation.DESTROY).isEqualTo(ee.getOperation());
-          assertThat("destroyCB" + i).isEqualTo(ee.getCallbackArgument());
-          assertThat(ee.isOriginRemote()).isTrue();
-        }
+
+      List<CacheEvent<Object, Object>> list = assertForOpCount(name, Operation.DESTROY, 10);
+
+      for (int i = 0; i < 10; i++) {
+        Object key = i;
+        EntryEvent ee = (EntryEvent) list.get(i);
+        assertThat(ee.getKey()).isEqualTo(key);
+        assertThat(ee.getOldValue()).isNull();
+        assertThat("destroyCB" + i).isEqualTo(ee.getCallbackArgument());
+        assertThat(ee.isOriginRemote()).isTrue();
       }
     });
+
     vm2.invoke("recreate", () -> {
       Region<Object, Object> region = getRootRegion().getSubregion(name);
       for (int i = 0; i < 10; i++) {
         Object key = i;
-        region.create(key, "create" + i);
+        region.create(key, "recreate" + i, "recreateCB" + i);
       }
     });
 
-    vm1.invoke("Verify creates", () -> {
+    vm1.invoke("Verify Local Load Creates", () -> {
       Region<Object, Object> region = getRootRegion().getSubregion(name);
-      CertifiableTestCacheListener<Object, Object> ctl =
-          (CertifiableTestCacheListener<Object, Object>) region.getAttributes()
-              .getCacheListeners()[0];
-      List<CacheEvent<Object, Object>> list = ctl.getEventHistory();
-      logger
-          .info("history (should be empty): " + list);
-      assertThat(list).isEmpty();
-      // now see if we can get it from the server
+
+      await().untilAsserted(() -> {
```

Comment:
This is important to wait so that the server has all the keys so that we don't get invalidates in addition to the Local_Load_Creates